File: problems/leetcode/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0;
        minPrice = prices[0];
        res = 0;
        for num in prices:
            if num > minPrice:
                if res < num - minPrice:
                    res = num - minPrice;
            else:
                minPrice = num;
        return res;

File: problems/leetcode/121.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = INT_MAX;
        int res = 0;
        for (int i = 0; i < prices.size(); i++) {
            if (prices[i] <= minPrice)
                minPrice = prices[i];
            else {
                if (res < prices[i] - minPrice)
                    res = prices[i] - minPrice;
            }
        }
        return res;
    }
};